### DISCUSSION TITLE
Need Help on Next Js 15 Server Actions

### DISCUSSION THREAD

**Discussion by adixoo** [vercel/next.js#74022] [Help]
### Summary

I'm developing a CRM Dashboard with two access modes: 'admin' and 'client', hosted on separate domains. To secure routes, I'm using environment variables to control access.

However, I'm encountering an issue with server actions located in a separate 'actions' folder. Since these actions are primarily designed for the admin panel, they are currently hosted even when the client-side application is running.

Given that the free tier of Vercel has a limited server action quota (approximately 10-12), I'm seeking a solution to conditionally host server actions. This would prevent exceeding the quota and ensure efficient resource utilization.

**Comment by eric-mxrtin** [comment]
Try to create a wrapper for your server actions that checks the current mode before executing:

```
// utils/serverActionWrapper.ts

import { NextResponse } from 'next/server';

export function withModeCheck(action: Function) {
  return async (...args: any[]) => {
    if (process.env.NEXT_PUBLIC_APP_MODE !== 'admin') {
      return NextResponse.json({ error: 'Unauthorized' }, { status: 403 });
    }
    return action(...args);
  };
}
```
